cmd/serve/restic/inproc.go

```diff
@@ -0,0 +1,21 @@
+package restic
+
+import (
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/rclone/rclone/cmd"
+	"github.com/rclone/rclone/cmd/serve/httplib/httpflags"
+)
+
+// NewServer returns a rclone server object which talks the restic http protocol
+func NewServer(args []string) (http.RoundTripper, error) {
```

Comment:
It would be best to return the specific type rather than the interface `http.RoundTripper` here - it leads to more flexibility later.

Reply:
golint fails if returned objects is of unexported types. So this is a workaround unless the server type is exported which would require more changes and I am not sure whether it is worthwhile.